internal/primitives/core/hashing/hashing.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package hashing
+
+import (
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/sha3"
+)
+
+// BlakeTwo256 returns a Blake2 256-bit hash of the input data
+func BlakeTwo256(data []byte) [32]byte {
```

Comment:
If I'm not mistaken don't we already have support for these?

Reply:
Yes but we don't return a fixed size array.  I will probably remove the other if we don't need it, or at the very least convert to this type when integrating with `db.Backend`.

Reply:
I think ideally we do not have multiple implementations of the same crypto primitive, but I can understand if this is something to address later. However, I do think it's something to avoid when the integration time comes

Reply:
see https://github.com/ChainSafe/gossamer/pull/4015#discussion_r1648179768.

Reply:
I misspoke, `common.Blake2bHash`does return a `common.Hash` which is a `[32]byte`.  I ended up introducing this because I really dislike the `common` package name, and I didn't want it to return an error, so more like a `MustBlake2bHash`.

Reply:
Okay that makes sense to me. I think it will be nice to just use one post-integration, but it seems like since the same type is returned that should be pretty easy to port over to this impl